pkg/ingester/ingester_v2.go

```diff
@@ -397,7 +406,12 @@ func (i *Ingester) getOrCreateTSDB(userID string, force bool) (*tsdb.DB, error)
 
 	// Create a new shipper for this database
 	if i.cfg.TSDBConfig.ShipInterval > 0 {
-		s := shipper.New(util.Logger, nil, udir, &Bucket{userID, i.TSDBState.bucket}, func() labels.Labels { return l }, metadata.ReceiveSource)
+		reg := prometheus.NewRegistry()
```

Comment:
If we have a custom struct for it, we could move this logic in a func like `NewRegistry(userID)` and then just call `shipper.New(..., x.NewRegistry(userID), ...)`.